### PR TITLE
T2 unlimited, launch template support

### PR DESCRIPTION
**Pull request by aermakov-zalando**
* Add support for launch templates. If the ASG uses a launch template, upgrade all nodes that don't use the same launch template version (this covers nodes started from a launch configuration, so upgrades are possible), otherwise use the old logic.
* Add `values.supports_t2_unlimited` that's set iff the instance type is T2.

### CODE REVIEW

File: provisioner/node_pools.go [hunk 1]
```diff
@@ -149,6 +149,7 @@ func (p *AWSNodePoolProvisioner) Provision(values map[string]interface{}) error
 
 // provisionNodePool provisions a single node pool.
 func (p *AWSNodePoolProvisioner) provisionNodePool(nodePool *api.NodePool, values map[string]interface{}) error {
+	values["supports_t2_unlimited"] = strings.HasPrefix(nodePool.InstanceType, "t2")
```
Comment (mikkeloscar) [inline provisioner/node_pools.go]: Why is this needed? Is the t2_unlimited special in any way?

File: pkg/updatestrategy/aws_asg.go [hunk 2]
```diff
@@ -448,50 +459,86 @@ func (n *ASGNodePoolsBackend) getInstancesToUpdate(asg *autoscaling.Group) (map[
...
-	launchConfig, err := n.getLaunchConfiguration(asg)
-	if err != nil {
-		return nil, err
-	}
+	launchParams := &asgLaunchParameters{}
 
-	oldInstances := make(map[string]bool)
+	if asg.LaunchTemplate != nil && aws.StringValue(asg.LaunchTemplate.LaunchTemplateName) != "" {
+		version := aws.StringValue(asg.LaunchTemplate.Version)
+		if version == "" || strings.HasPrefix(version, "$") {
```
[the remark this answers is not in the document]
Comment (aermakov-zalando) [reply pkg/updatestrategy/aws_asg.go]: You're right, I've added a comment.

**Comment by mikkeloscar** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:-1:

**Comment by mikkeloscar** [comment]
Waiting for cluster upgrades to finish :P
